### ISSUE TITLE
Compile issues with headers

### ISSUE THREAD

**Issue by dddejan** [CVC4/CVC4#3658]
I'm trying to compile a project using cvc4 and I'm getting the following error

```console
In file included from /usr/local/include/cvc4/cvc4.h:21:
/usr/local/include/cvc4/base/configuration.h:19:10: fatal error: 'cvc4_public.h' file not found
#include "cvc4_public.h"
         ^~~~~~~~~~~~~~~
```

The location of the header is `/usr/local/include/cvc4/cvc4_public.h`. This is with latest master after default `sudo make install`.

**Comment by mpreiner** [comment]
@dddejan can you give us more information? What's the output of `./configure.sh` and what OS are you using?

**Comment by dddejan** [comment]
```
-- The C compiler identification is AppleClang 11.0.0.11000033
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building Production build
-- Performing Test HAVE_FLAG_O3
-- Performing Test HAVE_FLAG_O3 - Success
-- Configuring with C flag '-O3'
-- Configuring with CXX flag '-O3'
-- Performing Test HAVE_FLAG_Wall
-- Performing Test HAVE_FLAG_Wall - Success
-- Configuring with C flag '-Wall'
-- Configuring with CXX flag '-Wall'
-- Performing Test HAVE_FLAG_fexceptions
-- Performing Test HAVE_FLAG_fexceptions - Success
-- Configuring with C flag '-fexceptions'
-- Performing Test HAVE_FLAG_Wno_deprecated
-- Performing Test HAVE_FLAG_Wno_deprecated - Success
-- Configuring with C flag '-Wno-deprecated'
-- Configuring with CXX flag '-Wno-deprecated'
-- Performing Test HAVE_FLAG_Wsuggest_override
-- Performing Test HAVE_FLAG_Wsuggest_override - Failed
-- Performing Test HAVE_FLAG_Wnon_virtual_dtor
-- Performing Test HAVE_FLAG_Wnon_virtual_dtor - Success
-- Configuring with CXX flag '-Wnon-virtual-dtor'
-- Performing Test HAVE_FLAG_Wimplicit_fallthrough
-- Performing Test HAVE_FLAG_Wimplicit_fallthrough - Success
-- Configuring with C flag '-Wimplicit-fallthrough'
-- Configuring with CXX flag '-Wimplicit-fallthrough'
-- Performing Test HAVE_FLAG_Wno_class_memaccess
-- Performing Test HAVE_FLAG_Wno_class_memaccess - Failed
-- Found PythonInterp: /usr/local/bin/python (found version "2.7.17") 
-- Found GMP: /usr/local/include  
-- Found GMP libs: /usr/local/lib/libgmp.dylib
-- Performing Test CVC4_NEED_INT64_T_OVERLOADS
-- Performing Test CVC4_NEED_INT64_T_OVERLOADS - Success
-- Performing Test CVC4_NEED_HASH_UINT64_T_OVERLOAD
-- Performing Test CVC4_NEED_HASH_UINT64_T_OVERLOAD - Failed
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for C++ include ext/stdio_filebuf.h
-- Looking for C++ include ext/stdio_filebuf.h - not found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for ffs
-- Looking for ffs - found
-- Looking for optreset
-- Looking for optreset - found
-- Looking for sigaltstack
-- Looking for sigaltstack - found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for strtok_r
-- Looking for strtok_r - found
-- Performing Test STRERROR_R_CHAR_P
-- Performing Test STRERROR_R_CHAR_P - Failed
-- Found Git: /usr/local/bin/git (found version "2.24.1") 
-- Looking for antlr3FileStreamNew
-- Looking for antlr3FileStreamNew - found
-- Found ANTLR: /Users/dejan/workspace/CVC4/deps/install/bin/antlr3  
-- Found ANTLR libs: /Users/dejan/workspace/CVC4/deps/install/lib/libantlr3c.a
-- Found Java: /Library/Java/JavaVirtualMachines/jdk-10.jdk/Contents/Home/bin/java (found version "10.0.0") found components:  Runtime 
CVC4 1.8-prerelease

Build profile        : production

GPL                  : off
Best configuration   : off
Optimized            : on
Optimization level   : off

Assertions           : off
Debug symbols        : off
Debug context mem mgr: off

Dumping              : on
Muzzle               : off
Proofs               : on
Replay               : off
Statistics           : on
Tracing              : off

ASan                 : off
UBSan                : off
TSan                 : off
Coverage (gcov)      : off
Profiling (gprof)    : off
Unit tests           : off
Valgrind             : off

Shared libs          : on
Static binary        : off
Java bindings        : off
Python bindings      : off
Python2              : off
Python3              : off

ABC                  : off
CaDiCaL              : off
CryptoMiniSat        : off
drat2er              : off
GLPK                 : off
LFSC                 : off
MP library           : gmp
Readline             : off
SymFPU               : off


CPPLAGS (-D...)      : NDEBUG CVC4_DUMPING CVC4_PROOF CVC4_STATISTICS_ON
CXXFLAGS             : -O3 -Wall -Wno-deprecated -Wnon-virtual-dtor -Wimplicit-fallthrough
CFLAGS               : -O3 -Wall -fexceptions -Wno-deprecated -Wimplicit-fallthrough

Install prefix       : /usr/local

CVC4 license         : modified BSD

Note that this configuration is NOT built against any GPL'ed libraries, so
it is covered by the (modified) BSD license.  This is, however, not the best
performing configuration of CVC4.  To build against GPL'ed libraries which
improve CVC4's performance, re-configure with '-DENABLE_GPL -DENABLE_BEST'.

Now change to 'build' and type make, followed by make check or make install.

-- Configuring done
-- Generating done
-- Build files have been written to: /Users/dejan/workspace/CVC4/build
```

**Comment by dddejan** [comment]
This is all on a mac.

The issue is that `cvc4_public.h` is installed in the base cvc4 include directory, but the `base/configuration.h` includes it with the local `"cvc4_public.h"` include.

**Comment by mpreiner** [comment]
Hmm interesting, when I do a `make install` (on Arch Linux) I have `#include <cvc4/cvc4_public.h>` in `cvc4/base/configuration.h`. The [fix-install-headers.sh](https://github.com/CVC4/CVC4/blob/master/src/fix-install-headers.sh) script should actually take care of fixing the header includes for `make install`. Do you get any error/warning while doing a `make install`?

**Comment by dddejan** [comment]
I did get some strange sed error

```console
$ make install
[  0%] Built target gen-theory
[  0%] Built target gen-tags-debug
[  0%] Built target gen-tags-trace
[  0%] Generating Debug_tags
[  1%] Generating Trace_tags
[  1%] Built target gen-tags
[  1%] Built target gen-gitinfo
[  4%] Built target gen-expr
[  4%] Generating options.cpp, options_holder.h, arith_options.cpp, arrays_options.cpp, base_options.cpp, booleans_options.cpp, builtin_options.cpp, bv_options.cpp, datatypes_options.cpp, decision_options.cpp, expr_options.cpp, fp_options.cpp, idl_options.cpp, main_options.cpp, parser_options.cpp, printer_options.cpp, proof_options.cpp, prop_options.cpp, quantifiers_options.cpp, sep_options.cpp, sets_options.cpp, smt_options.cpp, strings_options.cpp, theory_options.cpp, uf_options.cpp, arith_options.h, arrays_options.h, base_options.h, booleans_options.h, builtin_options.h, bv_options.h, datatypes_options.h, decision_options.h, expr_options.h, fp_options.h, idl_options.h, main_options.h, parser_options.h, printer_options.h, proof_options.h, prop_options.h, quantifiers_options.h, sep_options.h, sets_options.h, smt_options.h, strings_options.h, theory_options.h, uf_options.h
[  4%] Built target gen-options
[ 91%] Built target cvc4
[ 96%] Built target cvc4parser
[ 97%] Built target gen-tokens
[ 98%] Built target main
[100%] Built target cvc4-bin
[100%] Built target main-test
Install the project...
-- Install configuration: "Production"
-- Up-to-date: /usr/local/lib/cmake/CVC4/CVC4Targets.cmake
-- Up-to-date: /usr/local/lib/cmake/CVC4/CVC4Targets-production.cmake
-- Up-to-date: /usr/local/lib/cmake/CVC4/CVC4Config.cmake
-- Up-to-date: /usr/local/lib/cmake/CVC4/CVC4ConfigVersion.cmake
-- Up-to-date: /usr/local/share/cvc4/sat.plf
-- Up-to-date: /usr/local/share/cvc4/er.plf
-- Up-to-date: /usr/local/share/cvc4/smt.plf
-- Up-to-date: /usr/local/share/cvc4/th_base.plf
-- Up-to-date: /usr/local/share/cvc4/lrat.plf
-- Up-to-date: /usr/local/share/cvc4/drat.plf
-- Up-to-date: /usr/local/share/cvc4/th_arrays.plf
-- Up-to-date: /usr/local/share/cvc4/th_bv.plf
-- Up-to-date: /usr/local/share/cvc4/th_bv_bitblast.plf
-- Up-to-date: /usr/local/share/cvc4/th_bv_rewrites.plf
-- Up-to-date: /usr/local/share/cvc4/th_real.plf
-- Up-to-date: /usr/local/share/cvc4/th_int.plf
-- Up-to-date: /usr/local/share/cvc4/th_lra.plf
-- Up-to-date: /usr/local/share/man/man1/cvc4.1
-- Up-to-date: /usr/local/share/man/man5/cvc4.5
-- Up-to-date: /usr/local/share/man/man3/libcvc4.3
-- Up-to-date: /usr/local/share/man/man3/libcvc4parser.3
-- Up-to-date: /usr/local/share/man/man3/options.3cvc
-- Up-to-date: /usr/local/share/man/man3/SmtEngine.3cvc
-- Installing: /usr/local/lib/libcvc4.6.dylib
-- Up-to-date: /usr/local/lib/libcvc4.dylib
-- Up-to-date: /usr/local/include/cvc4/api/cvc4cpp.h
-- Up-to-date: /usr/local/include/cvc4/api/cvc4cppkind.h
-- Up-to-date: /usr/local/include/cvc4/base/configuration.h
-- Up-to-date: /usr/local/include/cvc4/base/exception.h
-- Up-to-date: /usr/local/include/cvc4/base/listener.h
-- Up-to-date: /usr/local/include/cvc4/base/modal_exception.h
-- Up-to-date: /usr/local/include/cvc4/context/cdhashmap_forward.h
-- Up-to-date: /usr/local/include/cvc4/context/cdhashset_forward.h
-- Up-to-date: /usr/local/include/cvc4/context/cdinsert_hashmap_forward.h
-- Up-to-date: /usr/local/include/cvc4/context/cdlist_forward.h
-- Up-to-date: /usr/local/include/cvc4/cvc4.h
-- Up-to-date: /usr/local/include/cvc4/cvc4_public.h
-- Up-to-date: /usr/local/include/cvc4/cvc4parser_public.h
-- Up-to-date: /usr/local/include/cvc4/expr/array.h
-- Up-to-date: /usr/local/include/cvc4/expr/array_store_all.h
-- Up-to-date: /usr/local/include/cvc4/expr/ascription_type.h
-- Up-to-date: /usr/local/include/cvc4/expr/chain.h
-- Up-to-date: /usr/local/include/cvc4/expr/datatype.h
-- Up-to-date: /usr/local/include/cvc4/expr/emptyset.h
-- Up-to-date: /usr/local/include/cvc4/expr/expr_iomanip.h
-- Up-to-date: /usr/local/include/cvc4/expr/expr_stream.h
-- Up-to-date: /usr/local/include/cvc4/expr/record.h
-- Up-to-date: /usr/local/include/cvc4/expr/symbol_table.h
-- Up-to-date: /usr/local/include/cvc4/expr/type.h
-- Up-to-date: /usr/local/include/cvc4/expr/uninterpreted_constant.h
-- Up-to-date: /usr/local/include/cvc4/expr/variable_type_map.h
-- Up-to-date: /usr/local/include/cvc4/expr/expr.h
-- Up-to-date: /usr/local/include/cvc4/expr/kind.h
-- Up-to-date: /usr/local/include/cvc4/expr/expr_manager.h
-- Up-to-date: /usr/local/include/cvc4/options/argument_extender.h
-- Up-to-date: /usr/local/include/cvc4/options/language.h
-- Up-to-date: /usr/local/include/cvc4/options/option_exception.h
-- Up-to-date: /usr/local/include/cvc4/options/options.h
-- Up-to-date: /usr/local/include/cvc4/options/printer_modes.h
-- Up-to-date: /usr/local/include/cvc4/options/set_language.h
-- Up-to-date: /usr/local/include/cvc4/parser/input.h
-- Up-to-date: /usr/local/include/cvc4/parser/parser.h
-- Up-to-date: /usr/local/include/cvc4/parser/parser_builder.h
-- Up-to-date: /usr/local/include/cvc4/parser/parser_exception.h
-- Up-to-date: /usr/local/include/cvc4/printer/sygus_print_callback.h
-- Up-to-date: /usr/local/include/cvc4/proof/unsat_core.h
-- Up-to-date: /usr/local/include/cvc4/smt/command.h
-- Up-to-date: /usr/local/include/cvc4/smt/logic_exception.h
-- Up-to-date: /usr/local/include/cvc4/smt/smt_engine.h
-- Up-to-date: /usr/local/include/cvc4/smt_util/lemma_channels.h
-- Up-to-date: /usr/local/include/cvc4/smt_util/lemma_input_channel.h
-- Up-to-date: /usr/local/include/cvc4/smt_util/lemma_output_channel.h
-- Up-to-date: /usr/local/include/cvc4/theory/logic_info.h
-- Up-to-date: /usr/local/include/cvc4/theory/theory_id.h
-- Up-to-date: /usr/local/include/cvc4/util/abstract_value.h
-- Up-to-date: /usr/local/include/cvc4/util/bitvector.h
-- Up-to-date: /usr/local/include/cvc4/util/bool.h
-- Up-to-date: /usr/local/include/cvc4/util/cardinality.h
-- Up-to-date: /usr/local/include/cvc4/util/divisible.h
-- Up-to-date: /usr/local/include/cvc4/util/gmp_util.h
-- Up-to-date: /usr/local/include/cvc4/util/hash.h
-- Up-to-date: /usr/local/include/cvc4/util/integer_cln_imp.h
-- Up-to-date: /usr/local/include/cvc4/util/integer_gmp_imp.h
-- Up-to-date: /usr/local/include/cvc4/util/maybe.h
-- Up-to-date: /usr/local/include/cvc4/util/proof.h
-- Up-to-date: /usr/local/include/cvc4/util/rational_cln_imp.h
-- Up-to-date: /usr/local/include/cvc4/util/rational_gmp_imp.h
-- Up-to-date: /usr/local/include/cvc4/util/regexp.h
-- Up-to-date: /usr/local/include/cvc4/util/resource_manager.h
-- Up-to-date: /usr/local/include/cvc4/util/result.h
-- Up-to-date: /usr/local/include/cvc4/util/sexpr.h
-- Up-to-date: /usr/local/include/cvc4/util/statistics.h
-- Up-to-date: /usr/local/include/cvc4/util/tuple.h
-- Up-to-date: /usr/local/include/cvc4/util/unsafe_interrupt_exception.h
-- Up-to-date: /usr/local/include/cvc4/util/floatingpoint.h
-- Up-to-date: /usr/local/include/cvc4/util/integer.h
-- Up-to-date: /usr/local/include/cvc4/util/rational.h
sed: 1: "/usr/local/include/cvc4 ...": extra characters at the end of l command
-- Installing: /usr/local/lib/libcvc4parser.6.dylib
-- Up-to-date: /usr/local/lib/libcvc4parser.dylib
-- Installing: /usr/local/bin/cvc4
```

**Comment by mpreiner** [comment]
Ok, see the issue. Can you change the fix-install-headers.sh to the following and check if this fixes it for you? I'll prepare a PR.

```
#!/bin/bash

dir=$1
find "$dir/include/cvc4/" -type f | \
  xargs sed -i '' 's/include.*"\(.*\)"/include <cvc4\/\1>/'
```

**Comment by dddejan** [comment]
Yup, that fixes it!

**Comment by 4tXJ7f** [comment]
@mpreiner fixed this in #3660.
